code/modules/mob/mob_grab.dm

```diff
@@ -440,8 +438,10 @@
 		affecting.pixel_y = 0 //used to be an animate, not quick enough for del'ing
 		affecting.layer = initial(affecting.layer)
 		affecting.grabbed_by -= src
+		if(affecting.eye_blind <= 3)
+			affecting.eye_blind = 0  //try to instantly clear blindness, if it was likely caused by us (may result in mild eye clearing but like it would wear off in 3 seconds anyways so)
 	qdel(hud)
-	return ..()
+	return QDEL_HINT_HARDDEL_NOW

```

Comment:
it's not about how it GC's, that's probably easy to fix
it's about how quickly it's gone and unable to be interacted with